Mark gpt-5 models as reasoning models

JDNdeveloper opened the pull request:
Fixes https://github.com/karthink/gptel/issues/1019

karthink commented:
@JDNdeveloper Thanks.  Should be `gpt-5 gpt-5-mini gpt-5-nano` and not `5 5-mini 5-nano`, shouldn't it?

JDNdeveloper commented:
> @JDNdeveloper Thanks. Should be `gpt-5 gpt-5-mini gpt-5-nano` and not `5 5-mini 5-nano`, shouldn't it?
> 
> EDIT: While you fix that please also split the list across two lines. It's not a hard rule but I try to wrap at <80 chars.

You're right, I had misread `o4-mini` as `4o-mini` and thought the `gpt-` wasn't needed. Fixed.

karthink commented:
@JDNdeveloper Thanks for the PR!
